plugin/src/software/aws/toolkits/eclipse/amazonq/lsp/AmazonQLspClientImpl.java

```diff
@@ -170,4 +178,92 @@ public final void ssoTokenChanged(final SsoTokenChangedParams params) {
             Activator.getLogger().error("Error processing " + kind + " ssoTokenChanged notification", ex);
         }
     }
+    public enum NotificationSeverity {
+        LOW, MEDIUM, HIGH
+    }
+
+    /**
+     * Shows a notification to the user with the specified content and severity.
+     * This method is designed to be overridden by subclasses that need to customize
+     * the notification behavior.
+     *
+     * @param params The notification parameters containing the message type and content
+     */
+    @Override
+    public final void showNotification(final NotificationParams params) {
+        Activator.getLogger().info("Received notification JSON: " + params.type().toString());
+        Display.getDefault().asyncExec(() -> {
+            String title = params.content().title() != null 
+                ? params.content().title() 
+                : "AWS Notification";
+            String message = params.content().text();
+            NotificationSeverity severity = determineNotificationSeverity(params.type());
+            handleNotification(severity, title, message);
+        });
+    }
```

Comment:
I'd like to see this logic moved out to its own utility - probably [here](https://github.com/aws/amazon-q-eclipse/blob/main/plugin/src/software/aws/toolkits/eclipse/amazonq/util/ToolkitNotification.java) or a separate utility that handles notification business logic rather than notification rendering.